apps/hub-app/src/components/SortDropdown.tsx

```diff
@@ -0,0 +1,66 @@
+import { MouseEvent } from 'react';
+import styled, { useTheme } from 'styled-components';
+import { AiOutlineCheck } from 'react-icons/ai';
+
+import { Button, Dropdown, Theme } from '@daohaus/ui';
+import { ParMd } from '@daohaus/ui';
+import { SORT_FIELDS } from '../utils/constants';
+import { BiChevronDown } from 'react-icons/bi';
+
+const Container = styled.div`
+  display: flex;
+  align-items: center;
+  gap: 2.4rem;
+  transform: translateX(-13.5rem);
+`;
+
+const DropdownButton = styled(Button)`
+  &.selected {
+    background-color: ${(props: { theme: Theme }) => props.theme.secondary};
+  }
+`;
+
+type FilterDropdownProps = {
+  sortBy: string;
+  toggleSortBy: (event: MouseEvent<HTMLButtonElement>) => void;
+};
+
+const SortDropdown = ({ sortBy, toggleSortBy }: FilterDropdownProps) => {
+  const theme = useTheme();
+
+  return (
+    <Container>
+      <ParMd>Sorted by</ParMd>
+      <Dropdown
+        align="end"
+        width="25rem"
+        bg={theme.button.secondary.bg}
+        trigger={
+          <Button secondary IconRight={BiChevronDown}>
+            {SORT_FIELDS[sortBy].name}
+          </Button>
+        }
+        items={Object.keys(SORT_FIELDS).map((key) => {
```

Comment:
this is really cool -- i imagine we'll use this type of approach in core ui for dropdowns